repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/keystonev2/multiextractregex/MultiExtractRegexTest.groovy

```diff
@@ -0,0 +1,87 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package features.filters.keystonev2.multiextractregex
+
+import org.openrepose.framework.test.ReposeValveTest
+import org.openrepose.framework.test.mocks.MockIdentityV2Service
+import org.rackspace.deproxy.Deproxy
+import org.rackspace.deproxy.MessageChain
+import spock.lang.Unroll
+
+import static javax.servlet.http.HttpServletResponse.SC_OK
+import static javax.servlet.http.HttpServletResponse.SC_UNAUTHORIZED
+
+class MultiExtractRegexTest extends ReposeValveTest {
+
+    def static originEndpoint
+    def static identityEndpoint
+
+    static MockIdentityV2Service fakeIdentityV2Service
+
+    def setupSpec() {
+
+        deproxy = new Deproxy()
+
+        def params = properties.defaultTemplateParams
+        repose.configurationProvider.applyConfigs("common", params)
+        repose.configurationProvider.applyConfigs("features/filters/keystonev2/common", params)
+        repose.configurationProvider.applyConfigs("features/filters/keystonev2/multiextractregex", params)
+        repose.start()
+
+        originEndpoint = deproxy.addEndpoint(properties.targetPort, 'origin service')
+        fakeIdentityV2Service = new MockIdentityV2Service(properties.identityPort, properties.targetPort)
+        identityEndpoint = deproxy.addEndpoint(properties.identityPort, 'identity service', null, fakeIdentityV2Service.handler)
+    }
+
+    def setup() {
+        fakeIdentityV2Service.resetHandlers()
+    }
+
+    @Unroll("#uriSegment")
+    def "When the tenant ID can be in multiple places in the request URI"() {
+        given: "a configured Identity service"
+        fakeIdentityV2Service.with {
+            client_token = UUID.randomUUID().toString()
+            client_tenantid = UUID.randomUUID().toString()
+        }
+
+        when:
+        "a request is sent to repose with URI containing $uriSegment"
+        MessageChain mc = deproxy.makeRequest(
+            url: "$reposeEndpoint/$uriSegment/${fakeIdentityV2Service.client_tenantid}",
+            method: 'GET',
+            headers: ['content-type': 'application/json', 'X-Auth-Token': fakeIdentityV2Service.client_token])
+
+        then: "everything with a valid Tenant ID that was extracted is sent to the origin service, otherwise it isn't"
+        mc.receivedResponse.code as Integer == serviceRespCode
+
+        if (serviceRespCode == SC_OK) {
+            assert mc.handlings.size() == 1
+            assert mc.handlings[0].request.headers.findAll("x-tenant-id").get(0).contains(fakeIdentityV2Service.client_tenantid)
+        } else assert mc.handlings.size() == 0
```

Comment:
Not a fan of branching assertions in tests. Not only does it make the expectation block more difficult to understand, but it means that the test name cannot be as descriptive since the test itself is covering more than one case. Would you be opposed to breaking the `serversBad` case into its own test? You do end up with some duplication that way, but I, personally, think that's preferred. Oh, and breaking that case out also lets you drop the `serviceRespCode` column and lets you unroll from a list (i.e., `uriSegment << ["serversOne", "serversTwo", "serversToo"]`) instead.